ansible-playbook tracer_demo.yml --diff
>>> Called __init__...
>>> Called v2_playbook_on_start...
[WARNING]: Could not match supplied host pattern, ignoring: foo
>>> Called v2_playbook_on_vars_prompt...
What is your username? [Admin]:

PLAY [Test __init__, v2_playbook_on_start, v2_playbook_on_vars_prompt, v2_playbook_on_play_start, and v2_playbook_on_no_hosts_matched methods] ******************************************
>>> Called v2_playbook_on_play_start...
skipping: no hosts matched
>>> Called v2_playbook_on_no_hosts_matched...

PLAY [Test v2_playbook_on_task_start, v2_runner_on_start, and v2_runner_on_ok methods] **************************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [Say hello] ************************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
ok: [managed_node_1] => {
    "msg": "Hello from the play, Admin!"
}
>>> Called v2_runner_on_ok...

TASK [Fail a task] **********************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
fatal: [managed_node_1]: FAILED! => {
    "msg": "This task fails."
}
...ignoring
>>> Called v2_runner_on_failed...

TASK [Skip a task] **********************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
skipping: [managed_node_1]
>>> Called v2_runner_on_skipped...

PLAY [Test v2_runner_on_unreachable method] *********************************************************************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [Try to ping an unreachable_node] **************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
fatal: [unreachable_node]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: Could not resolve hostname 999.999.999.999: Name or service not known", "unreachable": true}
...ignoring
>>> Called v2_runner_on_unreachable...

PLAY [Test v2_playbook_on_include method] ***********************************************************************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [Run tasks from an external file] **************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
>>> Called v2_runner_on_ok...
included: /home/rgarcia/repos/ansible-qrg/xtra/included_tasks.yml for managed_node_1
>>> Called v2_playbook_on_include...

TASK [Run an included task] *************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
ok: [managed_node_1] => {
    "msg": [
        "I am an included task."
    ]
}
>>> Called v2_runner_on_ok...

PLAY [Test v2_runner_retry method] ******************************************************************************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [Ping] *****************************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
[WARNING]: Platform linux on host managed_node_1 is using the discovered Python interpreter at /home/rgarcia/repos/ansible-qrg/.venv/bin/python3.12, but future installation of another
Python interpreter could change the meaning of that path. See https://docs.ansible.com/ansible-core/2.18/reference_appendices/interpreter_discovery.html for more information.
ok: [managed_node_1]
>>> Called v2_runner_on_ok...

TASK [Fail on purpose] ******************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
FAILED - RETRYING: [managed_node_1]: Fail on purpose (1 retries left).Result was: {
    "attempts": 1,
    "changed": false,
    "result": {
        "ansible_facts": {
            "discovered_interpreter_python": "/home/rgarcia/repos/ansible-qrg/.venv/bin/python3.12"
        },
        "changed": false,
        "failed": false,
        "ping": "pong",
        "warnings": [
            "Platform linux on host managed_node_1 is using the discovered Python interpreter at /home/rgarcia/repos/ansible-qrg/.venv/bin/python3.12, but future installation of another Python interpreter could change the meaning of that path. See https://docs.ansible.com/ansible-core/2.18/reference_appendices/interpreter_discovery.html for more information."
        ]
    },
    "retries": 2
}
>>> Called v2_runner_retry...
fatal: [managed_node_1]: FAILED! => {
    "attempts": 1,
    "result": {
        "ansible_facts": {
            "discovered_interpreter_python": "/home/rgarcia/repos/ansible-qrg/.venv/bin/python3.12"
        },
        "changed": false,
        "failed": false,
        "ping": "pong",
        "warnings": [
            "Platform linux on host managed_node_1 is using the discovered Python interpreter at /home/rgarcia/repos/ansible-qrg/.venv/bin/python3.12, but future installation of another Python interpreter could change the meaning of that path. See https://docs.ansible.com/ansible-core/2.18/reference_appendices/interpreter_discovery.html for more information."
        ]
    }
}
...ignoring
>>> Called v2_runner_on_failed...

PLAY [Test v2_runner_item_on_ok, v2_runner_item_on_failed, and v2_runner_item_on_skipped methods] ***************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [Loop over items] ******************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
>>> Called v2_on_file_diff...
ok: [managed_node_1] => (item=1) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": 1,
    "msg": "All assertions passed"
}
>>> Called v2_runner_item_on_ok...
>>> Called v2_on_file_diff...
ok: [managed_node_1] => (item=2) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": 2,
    "msg": "All assertions passed"
}
>>> Called v2_runner_item_on_ok...
failed: [managed_node_1] (item=3) => {
    "ansible_loop_var": "item",
    "assertion": "item <= 2",
    "changed": false,
    "evaluated_to": false,
    "item": 3,
    "msg": "Assertion failed"
}
>>> Called v2_runner_item_on_failed...
...ignoring
>>> Called v2_runner_on_failed...

TASK [Skip an item] *********************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
>>> Called v2_on_file_diff...
ok: [managed_node_1] => (item=1) => {
    "ansible_loop_var": "item",
    "item": 1
}
>>> Called v2_runner_item_on_ok...
skipping: [managed_node_1] => (item=2)
>>> Called v2_runner_item_on_skipped...
>>> Called v2_on_file_diff...
ok: [managed_node_1] => (item=3) => {
    "ansible_loop_var": "item",
    "item": 3
}
>>> Called v2_runner_item_on_ok...
>>> Called v2_runner_on_ok...

PLAY [Test v2_runner_on_async_poll and v2_runner_on_async_ok methods] *******************************************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [Successful poll] ******************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
ASYNC POLL on managed_node_1: jid=j183237709750.153450 started=1 finished=0
>>> Called v2_runner_on_async_poll...
ASYNC POLL on managed_node_1: jid=j183237709750.153450 started=1 finished=0
>>> Called v2_runner_on_async_poll...
ASYNC OK on managed_node_1: jid=j183237709750.153450
>>> Called v2_runner_on_async_ok...
ok: [managed_node_1]
>>> Called v2_runner_on_ok...

PLAY [Test v2_runner_on_async_failed method] ********************************************************************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [Unsuccessful poll] ****************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
ASYNC POLL on managed_node_1: jid=j616699183715.153564 started=1 finished=0
>>> Called v2_runner_on_async_poll...
ASYNC POLL on managed_node_1: jid=j616699183715.153564 started=1 finished=0
>>> Called v2_runner_on_async_poll...
ASYNC POLL on managed_node_1: jid=j616699183715.153564 started=1 finished=0
>>> Called v2_runner_on_async_poll...
ASYNC FAILED on managed_node_1: jid=j616699183715.153564
>>> Called v2_runner_on_async_failed...
fatal: [managed_node_1]: FAILED! => {"async_result": {"ansible_job_id": "j616699183715.153564", "finished": 0, "invocation": {"module_args": {"_async_dir": "/home/rgarcia/.ansible_async", "jid": "j616699183715.153564", "mode": "status"}}, "results_file": "/home/rgarcia/.ansible_async/j616699183715.153564", "started": 1, "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}, "changed": false, "msg": "async task did not complete within the requested time - 3s"}
...ignoring
>>> Called v2_runner_on_failed...

PLAY [Test v2_playbook_on_notify and v2_playbook_on_handler_task_start methods] *********************************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [Notify handler] *******************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
>>> Called v2_runner_on_start...
>>> Called v2_runner_on_start...
>>> Called v2_runner_on_start...
ok: [control_node] => {
    "msg": "Checking if I have to notify the handler..."
}
>>> Called v2_runner_on_ok...
changed: [managed_node_1] => {
    "msg": "Checking if I have to notify the handler..."
}
>>> Called v2_runner_on_ok...
ok: [managed_node_2] => {
    "msg": "Checking if I have to notify the handler..."
}
>>> Called v2_runner_on_ok...
ok: [unreachable_node] => {
    "msg": "Checking if I have to notify the handler..."
}
>>> Called v2_runner_on_ok...
>>> Called v2_playbook_on_notify...
Handler notified for managed_node_1.

RUNNING HANDLER [Run handler] ***********************************************************************************************************************************************************
>>> Called v2_playbook_on_handler_task_start...
Handler started.
>>> Called v2_runner_on_start...
ok: [managed_node_1] => {
    "msg": "Hello from the handler, Admin, on managed_node_1!"
}
>>> Called v2_runner_on_ok...

PLAY [Test v2_playbook_on_no_hosts_remaining] *******************************************************************************************************************************************
>>> Called v2_playbook_on_play_start...

TASK [This task will run] ***************************************************************************************************************************************************************
>>> Called v2_playbook_on_task_start...
>>> Called v2_runner_on_start...
>>> Called v2_runner_on_start...
>>> Called v2_runner_on_start...
>>> Called v2_runner_on_start...
ok: [control_node] => {
    "msg": "Running the first play..."
}
>>> Called v2_runner_on_ok...
fatal: [managed_node_1]: FAILED! => {
    "msg": "Running the first play..."
}
>>> Called v2_runner_on_failed...
ok: [managed_node_2] => {
    "msg": "Running the first play..."
}
>>> Called v2_runner_on_ok...
ok: [unreachable_node] => {
    "msg": "Running the first play..."
}
>>> Called v2_runner_on_ok...

NO MORE HOSTS LEFT **********************************************************************************************************************************************************************
>>> Called v2_playbook_on_no_hosts_remaining...

NO MORE HOSTS LEFT **********************************************************************************************************************************************************************
>>> Called v2_playbook_on_no_hosts_remaining...

PLAY RECAP ******************************************************************************************************************************************************************************
control_node               : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
managed_node_1             : ok=12   changed=1    unreachable=0    failed=1    skipped=1    rescued=0    ignored=4
managed_node_2             : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
unreachable_node           : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=1

>>> Called v2_playbook_on_stats...
